src/components/Header/AuthenticatedHeaderOption.js

```diff
@@ -0,0 +1,461 @@
+import React, { useState, useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { Link, NavLink } from "react-router-dom";
+import { PATHS, interpolatePath } from "../../constant";
+import { hasOneFrom } from "../../common/utils";
+import { actions as userActions } from "../User/redux/action";
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
+import ExpandLessIcon from "@mui/icons-material/ExpandLess";
+import useStyles from "./styles";
+import { DropDown, MobileDropDown } from "./DropDown";
+import { sendToken } from "../User/redux/api";
+import { actions as pathwayActions } from "../../components/PathwayCourse/redux/action";
+
+import {
+  Box,
+  IconButton,
+  Typography,
+  Menu,
+  Avatar,
+  MenuItem,
+} from "@mui/material";
+
+const rolesLandingPages = {
+  admin: PATHS.PARTNERS,
+  volunteer: PATHS.CLASS,
+  partner: PATHS.PARTNERS,
+};
+
+const SwitchView = ({
+  role,
+  setSwitchView,
+  handleCloseSwitchView,
+  switchView,
+}) => {
+  const classes = useStyles();
+
+  const roleLandingPage = rolesLandingPages[role];
+  return roleLandingPage ? (
+    <MenuItem
+      onClick={() => {
+        setSwitchView(role);
+        handleCloseSwitchView();
+      }}
+      sx={{ margin: "0px 10px" }}
+      className={switchView === role && classes.bgColor}
+    >
+      <NavLink to={roleLandingPage} className={classes.link}>
+        {role.charAt(0).toUpperCase() + role.slice(1)}
+      </NavLink>
+    </MenuItem>
+  ) : (
+    ""
+  );
+};
+
+function AuthenticatedHeaderOption({ toggleDrawer, leftDrawer }) {
+  const [partnerId, setPartnerId] = useState("");
```

Comment:
You can use the following as a temporary fix for https://github.com/navgurukul/bhanwari-devi/pull/290#issuecomment-1098792528 for the main menu options.  (The issue would still persist for the items that don't use this component such as Profile and Logout.)
```suggestion
  function HeaderNavLink(props) {
    return (
      <MenuItem
        onClick={toggleDrawer && toggleDrawer(false)}
        sx={{
          padding: 0,
          borderRadius: "8px"
        }}
      >
        <NavLink
          to={props.to}
          className={classes.link}
          activeClassName={classes.active}
          style={{ width: "100%" }}
        >
          <Typography
            variant="subtitle1"
            sx={{
              height: "36px",
              padding: "6px 16px",
              display: "flex",
              alignItems: "center",
              "&:hover": {
                backgroundColor: "#E9F5E9",
                borderRadius: "8px"
              }
            }}
          >
            {props.text}
          </Typography>
        </NavLink>
      </MenuItem>
    );
  }
  const [partnerId, setPartnerId] = useState("");
```